### PR TITLE
Add Zfinx spec

### PR DESCRIPTION
**Pull request by aswaterman**
This PR contributes a new chapter to the unprivileged spec that describes the Zfinx, Zhinx, and Zdinx extensions, which add floating-point instructions that operate on the x-registers.

### CODE REVIEW

File: src/zfinx.tex [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+\begin{commentary}
+Load-pair and store-pair instructions are not provided, so transferring
+double-precision operands in RV32Zdinx from or to memory requires
+two loads or stores.
+Register moves need only a single FSGNJ.D instruction, however.
+\end{commentary}
+
+\begin{commentary}
+In future, an RV64Zqinx quad-precision extension could be defined analogously
```
Comment (pdonahue-ventana) [inline src/zfinx.tex:115]:
```suggestion
In the future, an RV64Zqinx quad-precision extension could be defined analogously
```

File: src/zfinx.tex [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+{\tt mstatus} field FS is hardwired to 0 if the Zfinx extension is
+implemented, and FS no longer affects the trapping behavior of
+floating-point instructions or {\tt fcsr} accesses.
+
+The {\tt misa} bits F, D, and Q are hardwired to 0 when the Zfinx
+extension is implemented.
+
+\begin{commentary}
+A future discoverability mechanism might be used to probe the existence
+of the Zfinx, Zhinx, Zdinx, and Zqinx extensions.
```
Comment (pdonahue-ventana) [inline src/zfinx.tex:133]: I don't think that Zqinx should be included here.  Yes, this is talking about the future.  But I thought that it was only talking about the future with respect to the discoverability mechanism and not with respect to an extension that does not currently exist.

File: src/zfinx.tex (hunk 1 above)
Comment (aswaterman) [reply src/zfinx.tex:115]: I think the former is a correct Britishism, but yeah, don't see it in American English much.

File: src/zfinx.tex (hunk 2 above)
Comment (aswaterman) [reply src/zfinx.tex:133]: This is an editing error.  An earlier draft actually included a Zqinx spec.  Thanks for pointing it out.

**Comment by aswaterman** [comment]
This was manually merged a few weeks ago.

**Pull request closed without merging**